src/ghost-content/transparent.js

```diff
@@ -32,6 +33,21 @@ module.exports = ({ logger, projectLocation }) => {
       const filePath = path.join(folderPath, file)
       return fs.readFileAsync(filePath, 'utf-8')
     },
+
+    deleteFile: (folder, file) => {
+      const { folderPath } = normalizeFolder(folder)
+      const filePath = path.join(folderPath, file)
+      return Promise.fromCallback(cb => fs.unlink(filePath, cb))
+    },
+
+    directoryListing: (rootFolder, fileEndingPattern = '') => {
+      const flowsDir = normalizeFolder(rootFolder)
+      if (!fs.existsSync(flowsDir)) {
+        return Promise.resolve([])
+      }
+
+      return Promise.fromCallback(cb => glob(`**/*${fileEndingPattern}`, { cwd: flowsDir }, cb))
```

Comment:
An important thing to test in all scenarios (not saying it's currently wrong, just saying it needs to be checked): file names _must_ always be relative to the rootDir. For example, for a file `/Users/x/work/my-bot/flows/a/b/c.json` the `rootDir` is `flows`, and the file is `a/b/c.json`. This file must be named that way in all and every place.